### PR TITLE
Improve trip alert template (push + email)

### PR DESCRIPTION
**Pull request by AdrianaCeric**
### Checklist

- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

**Push**
Bullet point only present if >1 alert
image

**Email**
Also moves initial reminder ("Reminder for your upcoming trip...") outside of the list and to top of email
image

### CODE REVIEW

**Comment by JymDyerIBI** [comment]
I stuck a Javadoc comment on the unit test, to describe the magical behind-the-scenes snapshot file creation.

**Review by JymDyerIBI — APPROVED** [review]
👍  Code looks good to me.

**Comment by AdrianaCeric** [comment]
Thanks for the Javadoc @JymDyerIBI !

File: src/main/java/org/opentripplanner/middleware/models/TripMonitorNotification.java [hunk 1]
```diff
@@ -31,7 +41,6 @@ public TripMonitorNotification(NotificationType type, String body) {
     public int sortOrder() {
         return type == NotificationType.INITIAL_REMINDER ? -100 : 0;
     }
-
```
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/models/TripMonitorNotification.java]: Put this blank line back.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 2]
```diff
@@ -431,7 +431,10 @@ private void sendNotifications() {
             "notifications", notifications.stream()
```
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: This entry needs to be updated, it does not match `createTemplateRenderingTestCases` in the test. Try passing `notifications` as is (it is a `Set`), otherwise pass a `List` constructed from it.

File: src/test/java/org/opentripplanner/middleware/utils/TemplateUtilsTest.java [hunk 3]
```diff
@@ -50,7 +60,7 @@ public static List<TemplateRenderingTestCase> createTemplateRenderingTestCases()
         // Trip Monitor Notifications tests (for CheckMonitoredTrip#sendNotifications).
         Map<String, Object> notificationsData = Map.of(
             "tripId", "18f642d5-f7a8-475a-9469-800129e6c0b3",
-            "notifications", List.of("Test notification.", "Another notification.")
+            "notifications", List.of(new TripMonitorNotification(NotificationType.INITIAL_REMINDER, "This is the initial reminder text"), new TripMonitorNotification(NotificationType.DEPARTURE_DELAY, "This is the departure delay text"))
```
Comment (binh-dam-ibigroup) [inline src/test/java/org/opentripplanner/middleware/utils/TemplateUtilsTest.java]: Reformat placing each `new TripMonitorNotification(...)` on a new line with the correct indent.

File: src/test/java/org/opentripplanner/middleware/utils/TemplateUtilsTest.java [hunk 4]
```diff
@@ -102,4 +112,4 @@ private static <T>void assertMatchesSnapshot(String actual, String snapshotName)
         String actualWithStandardLineSeparators = actual.replaceAll("\r\n", "\n");
         assertThat(actualWithStandardLineSeparators, matchesSnapshot(snapshotName));
     }
-}
```
Comment (binh-dam-ibigroup) [inline src/test/java/org/opentripplanner/middleware/utils/TemplateUtilsTest.java]: Revert this change.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
One piece of code needs to be updated, and some formatting comments.

File: src/test/java/org/opentripplanner/middleware/utils/TemplateUtilsTest.java (hunk 4 above)
Comment (AdrianaCeric) [reply src/test/java/org/opentripplanner/middleware/utils/TemplateUtilsTest.java]: I'm not sure how to revert this change?

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java (hunk 2 above)
Comment (AdrianaCeric) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Let me know if my recent change is okay

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 5]
```diff
@@ -428,10 +428,13 @@ private void sendNotifications() {
...
-            "notifications", notifications.stream()
+            "notifications", new ArrayList<>(notifications).stream()
                 // Make certain notifications, such as the initial reminder one, appear on top.
                 .sorted(Comparator.comparingInt(TripMonitorNotification::sortOrder))
-                .map(notification -> notification.body)
+                .map(notification -> Map.of(
+                        "type", notification.type.toString(),
+                        "body", notification.body
+                ))
                 .collect(Collectors.toList())
```
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java]: Haha I think this should be simply:
```suggestion
            "notifications", new ArrayList<>(notifications)
```

**Review by binh-dam-ibigroup — APPROVED** [review]
Looks good, thanks for the changes!
